core/commands/object/object.go

```diff
@@ -391,15 +403,24 @@ And then run:
 			return
 		}
 
-		res.SetOutput(output)
+		if dopin {
+			n.Pinning.PinWithMode(objectCid, pin.Recursive)
+			err = n.Pinning.Flush()
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+		}
+
+		res.SetOutput(objectCid)
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			object := res.Output().(*Object)
-			return strings.NewReader("added " + object.Hash + "\n"), nil
+			object := res.Output().(*cid.Cid)
```

Comment:
We should keep this the same, and wrap the cid in an `Object` type when calling `SetOutput` above, otherwise we're changing the API. this is the last nitpick, i promise :)